File: LeetCode/Easy/How Many Numbers Are Smaller Than the Current Number/1365.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] arr = new int[nums.length];
        int[] cntarr = new int[101];
        
        int max = 0;
        for (int num : nums) {
            cntarr[num]++;
            if (num > max) {
                max = num;
            }
        }

        int sum = 0;
        for (int i = 0; i <= max; i++) {
            int currcnt = cntarr[i];
            cntarr[i] = sum;
            sum += currcnt;
        }

        for (int i = 0; i < nums.length; i++) {
            arr[i] = cntarr[nums[i]];
        }

        return arr;
    }
}

File: LeetCode/Easy/How Many Numbers Are Smaller Than the Current Number/1365.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
    const len = nums.length;
    const arr = new Array(len);
    const cntarr = new Array(101).fill(0);

    let max = 0;
    for (let i = 0; i < len; i++) {
        const num = nums[i];
        cntarr[num]++;
        if (num > max) {
            max = num;
        }
    }

    let sum = 0;
    for (let i = 0; i <= max; i++) {
        const currcnt = cntarr[i];
        cntarr[i] = sum;
        sum += currcnt;
    }

    for (let i = 0; i < len; i++) {
        arr[i] = cntarr[nums[i]];
    }

    return arr;
};
